dspace-server-webapp/src/main/java/org/dspace/app/rest/RestResourceController.java

```diff
@@ -1021,15 +1021,18 @@ public RepresentationModel listSearchMethods(@PathVariable String apiCategory, @
 
     /**
      * Internal method to convert the parameters provided as a MultivalueMap as a string to use in the self-link.
+     * This function will exclude all "embed" parameters
      * @param parameters
-     * @return encoded uriString containing request parameters
+     * @return encoded uriString containing request parameters without embed parameter
      */
     private String getEncodedParameterStringFromRequestParams(
             @RequestParam MultiValueMap<String, Object> parameters) {
         UriComponentsBuilder uriComponentsBuilder = UriComponentsBuilder.newInstance();
 
         for (String key : parameters.keySet()) {
-            uriComponentsBuilder.queryParam(key, parameters.get(key));
+            if (!StringUtils.equals(key, "embed")) {
```

Comment:
While I think this is correct based on current `main`, I think we're going to hit a "conflict" with this other open PR #3114 which adds an `embed.size` parameter.  I think we may need to consider loosening this `if` to make sure the param is not `embed` (exactly) or starts with `embed.`

If it makes it easier to test, you also could update your branch here with the changes in #3114... I'll leave that up to you though.

Reply:
@tdonohue Good looking out for this other PR that also touched upon this part of the code.

We've subsequently merged in PR #3114, and updated the code/tests to handle this additional case of embed.xxx params